### PR TITLE
docs: Updates to BACKUP and RESTORE diagrams

### PR DESCRIPTION
**Pull request by lnhsingh**
Release note: none

### CODE REVIEW

**Comment by lnhsingh** [comment]
@pbardea do you mind reviewing / do you know why tests are failing? Thanks!

**Comment by pbardea** [comment]
Yep, will take a look a bit later today. As to why the test is failing, the linter is complaining about some whilespace formatting in `pkg/cmd/docgen/diagrams.go`. I think running `./bin/crlfmt -w -tab=2 pkg/cmd/docgen/diagrams.go` from the base of the cockroach repo should apply the linting changes (that should run our formatter on the given file). Checking those changes in should fix CI.

**Comment by lnhsingh** [comment]
Thank you!

**Review by pbardea — COMMENTED** [review]
I'm not super familiar reading the docgen/diagrams.go is there a way to see the diagram they generate?

**Comment by lnhsingh** [comment]
How the Docs team previews the diagrams is by pasting the generated BNF into https://www.bottlecaps.de/rr/ui (in the Edit tab and then using the View Diagram tab to see the preview):

BACKUP BNF:
```
backup_stmt ::=
	'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' backup_options ( ( ',' backup_options ) )*
	| 'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' 'OPTIONS' '(' backup_options ( ( ',' backup_options ) )* ')'
	| 'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 
	| 'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination  'WITH' backup_options ( ( ',' backup_options ) )*
	| 'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination  'WITH' 'OPTIONS' '(' backup_options ( ( ',' backup_options ) )* ')'
	| 'BACKUP' targets 'INTO' ( | subdirectory | 'LATEST') 'IN' destination  
```

RESTORE BNF:
```
restore_stmt ::=
	'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' restore_options_list
	| 'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 
	| 'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' restore_options_list
	| 'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' restore_options_list
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' restore_options_list
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' restore_options_list
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' restore_options_list
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' restore_options_list
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 'AS' 'OF' 'SYSTEM' 'TIME' timestamp 
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' restore_options_list
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )  'WITH' 'OPTIONS' '(' restore_options_list ')'
	| 'RESTORE' ( 'TABLE' table_pattern ( ( ',' table_pattern ) )* | 'DATABASE' database_name ( ( ',' database_name ) )* ) 'FROM' subdirectory 'IN' ( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' ) 
```

File: pkg/cmd/docgen/diagrams.go [hunk 1]
```diff
@@ -441,21 +441,16 @@ var specs = []stmtSpec{
 	{
 		name:   "backup",
 		stmt:   "backup_stmt",
-		inline: []string{"table_pattern_list", "name_list", "opt_as_of_clause", "opt_incremental", "opt_with_options", "as_of_clause", "opt_with_backup_options"},
-		match:  []*regexp.Regexp{regexp.MustCompile("'BACKUP'")},
+		inline: []string{"opt_backup_targets", "opt_with_backup_options", "opt_as_of_clause", "as_of_clause", "backup_options_list"},
+		match:  []*regexp.Regexp{regexp.MustCompile("'BACKUP' targets 'INTO'")},
```
[the remark this answers is not in the document]
Comment (pbardea) [reply pkg/cmd/docgen/diagrams.go:445]: `RESTORE` explicitly lists out the targets (either tables or databases). Is that something we want to do here for consistency?

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62032)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea)

___
*[pkg/cmd/docgen/diagrams.go, line 445 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/62032#-MVrSP9a1vTPrbdEkH8E-r3-445:-MXE6xsD9dagjh4CUuz0:b-bw8e1h) ([raw file](https://github.com/cockroachdb/cockroach/blob/f15a251ef7103400ddc055738978f4eca0155e69/pkg/cmd/docgen/diagrams.go#L445)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

`RESTORE` explicitly lists out the targets (either tables or databases). Is that something we want to do here for consistency?
</blockquote></details>

Updated!
___
*[pkg/cmd/docgen/diagrams.go, line 448 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62032#-MVrSP9a1vTPrbdEkH8E-r4-448:-MXE3zzC--i0vBud0wVp:b-bw8e14) ([raw file](https://github.com/cockroachdb/cockroach/blob/728da3a4ed6d6eb314ef2571e2f61371dcd7db4a/pkg/cmd/docgen/diagrams.go#L448)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

I think we also need to make the entire `(subdir, latest) in` clause optional too:
```
"'INTO' (( | subdirectory | 'LATEST') 'IN')"
```

This would be to allow both of these statements:
```
BACKUP INTO 'userfile://my.files/my_collection'; -- full backup
BACKUP INTO LATEST IN 'userfile://my.files/my_collection'; -- incremental backup
```
</blockquote></details>

Updated.
___
*[pkg/cmd/docgen/diagrams.go, line 450 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/62032#-MVrSP9a1vTPrbdEkH8E-r4-450:-MXE4m_ZBRDZyJjE2QR6:b-2zbyx2) ([raw file](https://github.com/cockroachdb/cockroach/blob/728da3a4ed6d6eb314ef2571e2f61371dcd7db4a/pkg/cmd/docgen/diagrams.go#L450)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

The restore diagram lists  `( destination | '(' partitioned_backup_location ( ',' partitioned_backup_location )* ')' )" as the destination, noting the the backup location may also be partitioned. Should we do the same here for consistency?
</blockquote></details>

Yeah, good call. Updated.

**Review by pbardea — APPROVED** [review]
LGTM!

**Pull request merged**